packages/block-library/src/post-content/edit.js

```diff
@@ -45,12 +49,28 @@ function EditableContent( { context = {} } ) {
 		{ id: postId }
 	);
 
+	const entityRecord = useSelect(
+		( select ) => {
+			return select( coreStore ).getEntityRecord(
+				'postType',
+				postType,
+				postId
+			);
+		},
+		[ postType, postId ]
+	);
+
+	const hasInnerBlocks = !! entityRecord?.content?.raw;
+
+	const initialInnerBlocks = [ [ 'core/paragraph' ] ];
+
 	const props = useInnerBlocksProps(
 		useBlockProps( { className: 'entry-content' } ),
 		{
-			value: blocks,
+			value: hasInnerBlocks ? blocks : undefined,
```

Comment:
Ok for some reason, `useEntityBlockEditor` returns an empty array for `blocks` when the empty page is loaded after a non-empty page, but returns a single block if the empty page is loaded initially. I haven't dug much further - I'm guessing there's some bug in `getEditedEntityRecord` from which this originates. 

The problem with `blocks` being an empty array is that `innerBlocksProps.value` evaluates to true [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inner-blocks/index.js#L222), which causes the whole thing to render as `ControlledInnerBlocks`, which returns empty because there are, in fact, no blocks 😅 

So this change fixes that immediate issue, anyway. I think any further exploration is out of scope for this PR.

Reply:
(I guess an alternative/complementary fix would be to change the check in useInnerBlocksProps to `innerBlocksProps.value?.length && innerBlocksProps.onChange`)